sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/FilterPushdownBenchmark.scala

```diff
@@ -135,6 +139,34 @@ object FilterPushdownBenchmark extends BenchmarkBase with SQLHelper {
     benchmark.run()
   }
 
+  def filterPushDownBenchmarkWithColumn(
```

Comment:
a general note: I think `FilterPushdownBenchmark` is to test how good are these file formats about skipping disk IO with pushed filter. Checking the performance of building the filters may go beyond its scope.

Maybe we can just do a manual microbenchmark, and post the benchmark and result in PR description.

cc @dongjoon-hyun

Reply:
I'm fine with getting rid of it. It was useful for me during development because I wanted to make sure the version I implement is actually linear (since there were some tricky moments). However, it doesn't really matter that much in practice if there are no regressions in the other benchmarks.

Reply:
I think we should get rid of the changes in this file. 

However, we need a typical test case/query in PR description for verifying the perf gain.

Reply:
I think we should definitely have some automated benchmark for this. Otherwise there's nothing in the codebase exercising the behaviour being changed, and so nothing to prevent future regressions.

I didn't get initially that the proposal is to remove both new benchmarks here. I'm fine with removing the one that only does the predicate conversion (`"Predicate conversion benchmark with unbalanced Column"`). However, I'm strongly opposed to removing both as it would then be too easy to re-introduce the perf issue. This is a perf area that's already had to be fixed / worked around before (see the PR introducing the `buildTree` function to avoid skewed trees, for example) so I would feel uncomfortable leaving it without automated coverage.

If there's a more appropriate place to put the remaining benchmark (for example, `OrcV1FilterSuite`), I'm completely open to moving it. However, I'm under the impression from other code reviews that the regular test suites in Spark are not meant to test performance, and this would go against that.

Reply:
@IvanVergiliev . The following doesn't mean put that into here. Since this contribution is big, it's worth to have its own benchmark focusing on filter conversion.
> I think we should definitely have some automated benchmark for this. Otherwise there's nothing in the codebase exercising the behaviour being changed, and so nothing to prevent future regressions.